docs/content/guides/non-argument-deps.mdx

```diff
@@ -0,0 +1,136 @@
+
+### An unzipped CSV of stopwords
+
+Let's define an asset that contains the unzipped contents of our zip file:
+
+```python
+import zipfile
+
+@asset(non_argument_deps={"stopwords_zip"})
+def stopwords_csv() -> None:
+    with zipfile.ZipFile("stopwords.zip", "r") as zip_ref:
+        zip_ref.extractall(".")
+```
+
+The implementation extracts our zip file to a CSV file in the same directory. It looks a lot like the assets we defined earlier in this tutorial, with one main difference.
+
+In the assets we've seen so far, we indicate the asset dependencies by including function arguments with the name of the upstream assets. Dagster then invokes an I/O manager to load the asset as a Python object and supply it to our function.
+
+However, in this case, we don't want a Python object for the upstream asset, we just want to work with the file directly. So we instead supply `"stopwords_zip"` to the `non_argument_deps` parameter of the <PyObject module="dagster" object="asset" displayText="@asset"/> decorator. This lets us tell Dagster that the `stopwords` asset depends on the `stopwords_zip` asset, without telling it to load `stopwords_zip` for us.
+
+### Nabisco cereal ratings
+
+Finally, let's define an asset that joins the cereal ratings data from `stopwords` with the cereals in our `topstories_word_cloud` asset:
+
+```python
+import csv
+
+@asset(non_argument_deps={"stopwords_csv"})
+def topstories_word_cloud(topstories):
+    with open("stopwords.csv", "r") as f:
+        stopwords = {row[0] for row in csv.reader(f)}
+
+    titles_text = " ".join([str(item) for item in topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(
+        titles_text
+    )
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Convert the image to a saveable format
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    # Convert the image to Markdown to preview it within Dagster
+    md_content = f"![img](data:image/png;base64,{image_data.decode()})"
+
+    # Attach the Markdown content as metadata to the asset
+    return Output(
+        value=image_data,
+        metadata={
+            "plot": MetadataValue.md(md_content)
+        }
+    )
+
+    return result
+```
+
+This asset mixes and matches argument-based dependencies with non-argument-based dependencies. It depends on the `topstories_word_cloud` asset, which we defined in the previous section of the tutorial, and it relies on Dagster to load that asset's value into memory. It also depends on the `stopwords` asset, and the decorated function takes responsibility for loading that asset's value into memory.
```

Comment:
> It depends on the `topstories_word_cloud` asset, which we defined in the previous section of the tutorial,

🔧 This isn't part of the tutorial, so this sentence isn't accurate. Should it point right to the section of the tutorial where the user does this?